mgr [progress] [autoscaler]: Fix Autoscaler progress completion + progress module refactors

kamoltat opened the pull request:
**Summary**

This PR fixes an issue where the PG Autoscaler progress event could complete incorrectly due to comparing against the wrong target, and includes readability refactors in the progress module along with new test coverage.

**Problem**

The Autoscaler event’s completion check compared:

`pool_data["pg_num"] == pool_data["pg_num_target"]`

However, pool_data["pg_num_target"] is not the current scaling target; it reflects the previously computed value (i.e., what the pool was expected to have already reached). As a result, the comparison could erroneously evaluate true and prematurely complete the Autoscaler progress event.

**Fix**

Use the event’s target recorded at creation time:

`pool_data["pg_num"] == ev.pg_num_target`

This ensures the event only completes when the pool’s pg_num actually reaches the intended target for this scaling action.

**Additional progress-module cleanup**

Rename def _refresh() → def _persist() to better reflect behavior.

Rename variables (e.g., pg_stats_active_clean) to more intuitive names for readability.

Refactor global_event_update_progress() for clearer flow and maintainability.

**Tests**

Add unit tests for the PgAutoscaler event logic in pybind/mgr/progress/test_progress.py to cover correct completion behavior.

Fixes: https://tracker.ceph.com/issues/72857

<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)
</details>

kamoltat committed:
pybind/mgr/pg_autoscaler: fix progress event of the autoscaler

Problem:
We are comparing the wrong value when determining whether the Autoscaler event should be completed or not,
we compared pool_data["pg_num"] == pool_data["pg_num_target"] which will always return true since pool_data["pg_num_target"]
is not actual the target pg we are trying to scale towards, but rather the previous calculated value that we
are expected to already reached based on previous values.

Solution:
Use the correct value which is comparing pool_data[‘pg_num’] == ev.pg_num_target.

Fixes: https://tracker.ceph.com/issues/72857

Signed-off-by: Kamoltat Sirivadhna <ksirivad@redhat.com>

kamoltat committed:
pybind/mgr/progress: refractor Global Recovery Event

1. rename def _refresh() to def _persist() for _persist is more true to what the function actually does.
2. rename pg_stats_active_clean Active Py Module objects such that it is more intuitive and more readable.
3. refractor def global_event_update_progres for better readability.

Fixes: https://tracker.ceph.com/issues/72857

Signed-off-by: Kamoltat Sirivadhna <ksirivad@redhat.com>

Conflicts:
	src/pybind/mgr/progress/module.py - Trivial Fix

kamoltat committed:
pybind/mgr/progress/test_progress.py: Add PgAutoscaler event test coverage

Fixes: https://tracker.ceph.com/issues/72857

Signed-off-by: Kamoltat Sirivadhna <ksirivad@redhat.com>

kamoltat commented:
jenkins test make check arm64

ronen-fr commented on src/pybind/mgr/progress/module.py:
```diff
@@ -39,10 +39,13 @@ def __init__(self, id: str,
         self._add_to_ceph_s = add_to_ceph_s
 
-    def _refresh(self):
+    def _persists(self) -> None:
```
nit: 'refresh' is in a different tense than 'persists'

ronen-fr reviewed (approved):
LGTM (some nits)

kamoltat commented:
@ronen-fr Thank you so much for the review!
I made the changes per your comments. Moving this to `needs-qa`. Let me know if you have other concerns.

kamoltat committed:
doc/placement-groups.rst: bias = pg_autoscale_bias config value

make it more obvious for the reader that when setting `bias`
in the autoscaler, they need to type in `pg_autoscale_bias`

Fixes: https://tracker.ceph.com/issues/72857

Signed-off-by: Kamoltat Sirivadhna <ksirivad@redhat.com>

bluikko commented:
The "readthedocs" action fails because of Python package changes at RTD. This was fixed for `main` in #66787 and the action should progress if you are able to rebase your branch onto a fresh `main` that includes commit 15481e509b4d644d0644188501d86a4ceda2c039.
